File: recursion/recursion-basics/sum_to_n.cs
using System;

namespace isaac_code_samples
{
    class SumToNRecursion
    {
        static void Main(string[] args){
            int n = 6;
            int result = SumToN(n);
            Console.WriteLine(String.Format("The sum of 1 to {0} is: {1}", n.ToString(), result.ToString()));
        }

        public static int SumToN(int n){
            //returns the sum of all natural numbers from 1 to n inclusive
            if (n == 1){
                return 1;
            }else{
                return n + SumToN(n-1);
            }
        }
    }
}

File: recursion/recursion-basics/python/sum_to_n.py
# Raspberry Pi Foundation
# Developed to be used alongside Isaac Computer Science,
# part of the National Centre for Computing Education
# Usage licensed under CC BY-SA 4

def sum_to_n(n):
    """Recursively calculates and returns the sum of all natural numbers from 1 to n inclusive"""
    if n == 1:
        return 1
    else:
        return n + sum_to_n(n-1)


# This code will run if this file is executed directly
# (i.e. not called by another program)    
if __name__ =='__main__':
    n = 6
    result = sum_to_n(n)
    print(f"The sum of 1 to {n} is: {result}")
